src/main/java/kr/codesquad/library/controller/AccountController.java

```diff
@@ -1,32 +1,57 @@
 package kr.codesquad.library.controller;
 
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
 import kr.codesquad.library.domain.account.response.AccountMyPageResponse;
 import kr.codesquad.library.domain.account.response.AccountProfileResponse;
+import kr.codesquad.library.domain.account.response.RoleRequestResponse;
 import kr.codesquad.library.global.api.ApiResult;
 import kr.codesquad.library.global.config.oauth.dto.AccountPrincipal;
 import kr.codesquad.library.global.config.resolver.LoginAccount;
+import kr.codesquad.library.global.error.ErrorCode;
+import kr.codesquad.library.global.error.ErrorResponse;
+import kr.codesquad.library.global.error.exception.domain.BadRequestAuthorizationException;
 import kr.codesquad.library.service.AccountService;
 import lombok.RequiredArgsConstructor;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
-import static kr.codesquad.library.global.api.ApiResult.*;
+import static kr.codesquad.library.global.api.ApiResult.OK;
 
+@Slf4j
+@Api
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("/v1/users")
 public class AccountController {
 
     private final AccountService accountService;
 
+    @ApiOperation(value = "마이페이지")
     @GetMapping("/mypage")
     public ApiResult<AccountMyPageResponse> getMyPage(@LoginAccount AccountPrincipal loginAccount) {
         return OK(accountService.getMyPage(loginAccount.getId()));
     }
 
+    @ApiOperation(value = "프로필")
     @GetMapping("/profile")
     public ApiResult<AccountProfileResponse> getProfile(@LoginAccount AccountPrincipal loginAccount) {
         return OK(accountService.getProfile(loginAccount.getId()));
     }
+
+    @ApiOperation(value = "USER 권한 요청")
+    @PutMapping()
+    public ApiResult<RoleRequestResponse> requestAuthority(@LoginAccount AccountPrincipal loginAccount) {
```

Comment:
처음보는 기능들이 있는데 아래 기능들은 어떤 역할인가요?? 

- [ ] @Api 어노테이션의 기능?
- [ ] @loginaccount의 기능은?
- [ ] @ApiResult의 기능은?

Reply:
- `@Api`, `@ApiOperation`은 swagger 문서에 표시하는 용도로 없어도 되는 기능입니다.
- `@LoginAccount` 는 이렇게 만든 애노테이션입니다.
```suggestion
  @Target(ElementType.PARAMETER)
  @Retention(RetentionPolicy.RUNTIME)
  @AuthenticationPrincipal
  public @interface LoginAccount {
  }
```
여기서 확인해야할 건 바로 `@AuthenticationPrincipal` 입니다. 이 애노테이션의 역할은 시큐리티와 관계가 있습니다. 애노테이션을 붙이면 이 애노테이션으로 유저의 인증, 인가를 판단할 수 있습니다. [이동욱님 책](http://www.yes24.com/Product/Goods/83849117?OzSrank=2)이나 [개장모](https://github.com/gae-jang-mo/app) 에선 resolver를 만들어 빈으로 등록하는 등 여러 방법을 써야하지만 Security 4.0 부터 들어온 이 애노테이션으로 단 한줄로 처리가 가능합니다. 

이게 더 유효한가 아닌가는 아직까지 제대로 판단을 하지 못하겠습니다. 제가 두개 다 구현해봤는데 역할이 크게 다르지 않아서 저는 이대로 `@LoginAccount`를 사용했습니다.

Reply:
`ApiResult`는 팀내에서 합의한 JSON 데이터 응답(response) dto 입니다. 200일 경우에만 전달이 되도록 설정했으며, status, statusCode, statusMessage, data로 만들어졌습니다. 프론트에서 만약 정확하게 왔으면 굳이 head에서 확인이 아닌 JSON에서 확인하고 싶다고 하셔서 만들었습니다. 

만약 프론트분께서 그런 설명없으셨으면 그대로 ResponseEntity를 썼을거에요